pdr_backend/lake/table_bronze_pdr_predictions.py

```diff
@@ -0,0 +1,187 @@
+from typing import Dict
+
+import polars as pl
+from enforce_typing import enforce_types
+from polars import Boolean, Float64, Int64, Utf8
+
+from pdr_backend.lake.plutil import (
+    pick_df_and_ids_on_period,
+)
+from pdr_backend.ppss.ppss import PPSS
+
+
+bronze_pdr_predictions_table_name = "bronze_pdr_predictions"
+
+# CLEAN & ENRICHED PREDICTOOR PREDICTIONS SCHEMA
+bronze_pdr_predictions_schema = {
+    "ID": Utf8,  # f"{contract}-{slot}-{user}"
+    "slot_id": Utf8,  # f"{contract}-{slot}"
+    "contract": Utf8,  # f"{contract}"
+    "slot": Int64,
+    "user": Utf8,
+    "pair": Utf8,
+    "timeframe": Utf8,
+    "source": Utf8,
+    "predvalue": Boolean,
+    "truevalue": Boolean,
+    "stake": Float64,
+    "payout": Float64,
+    "timestamp": Int64,
+    "last_event_timestamp": Int64,
+}
+
+
+def _process_predictions(
+    collision_ids: pl.Series, dfs: Dict[str, pl.DataFrame], ppss: PPSS
+) -> Dict[str, pl.DataFrame]:
+    """
+    @description
+        Perform post-fetch processing on the data.
+        1. Find predictions within the update
+        2. Transform predictions to bronze
+        3. Concat to existing table
+    """
+
+    # only add new predictions
+    predictions_df = dfs["pdr_predictions"].filter(
+        (pl.col("timestamp") >= ppss.lake_ss.st_timestamp)
+        & (pl.col("timestamp") <= ppss.lake_ss.fin_timestamp)
+        & (pl.col("ID").is_in(collision_ids).not_())
+    )
```

Comment:
Implementation looks good, an improvement we could have in the future is to only update the existing columns and add new rows whenever new data is fetched from subgraph instead of recreating the entire bronz table each time new data is fetched

Reply:
Great, I agree! 
I added some logic (1) to reduce how many records need to be modified but this can be improved further.

I have also (2) shared latest on how to avoid recreating the bronze table every time.